subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/JacocoPluginIntegrationTest.groovy

```diff
@@ -144,5 +144,159 @@ class JacocoPluginIntegrationTest extends AbstractIntegrationSpec {
+
+            Artifacts
+                - build${File.separator}jacoco${File.separator}test.exec (artifactType = binary)""".stripIndent())
+    }
+
+    @ToBeFixedForConfigurationCache(because = ":outgoingVariants")
+    def "jacoco plugin adds outgoing variants for custom test suite"() {
+        buildFile << """
+            testing {
+                suites {
+                    integrationTest(JvmTestSuite) {
+                        testType = TestType.INTEGRATION_TESTS
+
+                        dependencies {
+                            implementation project
+                        }
+                    }
+                }
+            }
+        """.stripIndent()
+
+        expect:
+        succeeds "outgoingVariants"
+
+        outputContains("""
+            --------------------------------------------------
+            Variant coverageDataElementsForIntegrationTest
+            --------------------------------------------------
+            Capabilities
+                - :${getTestDirectory().getName()}:unspecified (default capability)
+            Attributes
+                - org.gradle.category      = documentation
+                - org.gradle.docstype      = jacoco-coverage-bin
+                - org.gradle.targetname    = integrationTest
+                - org.gradle.testsuitename = integrationTest
+                - org.gradle.testsuitetype = integration-tests
+                - org.gradle.usage         = verification
+
+            Artifacts
+                - build${File.separator}jacoco${File.separator}integrationTest.exec (artifactType = binary)""".stripIndent())
+    }
+
+    def "Jacoco coverage data can be consumed by another task via Dependency Management"() {
+        buildFile << """
+            test {
+                jacoco {
+                    destinationFile = layout.buildDirectory.file("jacoco/jacocoData.exec").get().asFile
+                }
+            }
+            """.stripIndent()
+
+        buildFile << """
+            // A resolvable configuration to collect JaCoCo coverage data
+            def coverageDataConfig = configurations.create("coverageData") {
+                visible = true
+                canBeResolved = true
+                canBeConsumed = false
+                attributes {
+                    attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.VERIFICATION))
+                    attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, Category.DOCUMENTATION))
+                    attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named(DocsType, DocsType.JACOCO_COVERAGE))
+                }
+            }
+
+            dependencies {
+                coverageData project
+            }
+
+            def testResolve = tasks.register('testResolve') {
+                doLast {
+                    assert coverageDataConfig.getResolvedConfiguration().getFiles()*.getName() == [test.jacoco.destinationFile.name]
```

Comment:
Nit: be groovier using property syntax: `assert coverageDataConfig.resolvedConfiguration.files*.name == [test.jacoco.destinationFile.name]`